core/probe.cpp

```diff
@@ -396,6 +398,11 @@ void Probe::delayedInit()
         showInProcessUi();
 }
 
+void Probe::shutdown()
+{
+    delete this;
```

Comment:
yikes ;-) usually, this is a very bad idea, but here there's really no way around this, is there?